src/index.js

```diff
@@ -3,16 +3,27 @@ export default middleware
 
 
 export { runSaga } from './internal/runSaga'
-export { CANCEL } from './internal/proc'
 export { END, eventChannel, channel } from './internal/channel'
 export { buffers } from './internal/buffers'
 export { takeEvery, takeLatest } from './internal/sagaHelpers'
-export { delay } from './internal/utils'
 
+import { CANCEL } from './internal/proc'
 import * as effects from './effects'
 import * as utils from './utils'
 
+export function delay(ms, val=true) {
+  let timeoutId;
+  const promise = new Promise((resolve) => {
+    timeoutId = setTimeout(resolve, ms, val);
```

Comment:
@aikoven TBH didn't thank of browser compat. But in this simple case I think we could call the 'cross-browser' version of setTimeout (without args...)